DAQ_System/Source/Adapter/DataLogger/SdDataLogger.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+* SD Data Logger
+* Authors:        Cristian Cruz
+*                 Ethan Fischer
+*
+* Email:          cris14.alex@gmail.com
+*                 ejf180001@utdallas.edu
+* 
+* (c) 2022 Dallas Formula Racing - Embedded Firmware Team
+* Formula SAE International Collegiate Chapter
+* GPL-3.0 License
+*/
+
+#include "SdDataLogger.hpp"
+#include "FATFileSystem.h"
+#include "SDBlockDevice.h"
+
+namespace adapter {
+
+// used to set a new file name for each new log session
+int log_session = 0;
+char file_name[16] = "\0";
+
+// block device and file system declarations
+SDBlockDevice block_device;
+FATFileSystem file_system;
+
+SdDataLogger::SdDataLogger(PinName mosi, PinName miso, PinName sck, PinName cs) : kBlockSectorByteSize(512) {
+    SDBlockDevice block_device(mosi, miso, sck, cs);
+    FATFileSystem file_system("fs");
+    Mount(&file_system, &block_device);
+
+    // initialize write buffer used to write to a file
+    char write_buffer[kBlockSectorByteSize];
```

Comment:
- Remove `kBlockSectorByteSize` from the initializer list
- Add the `block_device` to the initializer list. For formatting, I suggest (note the two spaces before the colon):

```c++
SdDataLogger::SdDataLogger(PinName mosi, PinName miso, PinName sck, PinName cs)
  : block_device(mosi, miso, sck, cs) {

```

- Initialize `file_system` with "fs" in its declaration with the header file. This is the only file system we will need from Mbed

Reply:
Done